src/modules/launcher/Plugins/Microsoft.Plugin.Registry.UnitTest/Helper/RegistryHelperTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Linq;
+using Microsoft.Plugin.Registry.Helper;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Plugin.Registry.UnitTest.Helper
+{
+    [TestClass]
+    public sealed class RegistryHelperTest
+    {
+        [TestMethod]
+        [DataRow(@"HKCC\System\CurrentControlSet\Control", "HKEY_CURRENT_CONFIG")]
+        [DataRow(@"HKCR\*\OpenWithList", "HKEY_CLASSES_ROOT")]
+        [DataRow(@"HKCU\Control Panel\Accessibility", "HKEY_CURRENT_USER")]
+        [DataRow(@"HKLM\HARDWARE\UEFI", "HKEY_LOCAL_MACHINE")]
+        [DataRow(@"HKPD\???", "HKEY_PERFORMANCE_DATA")]
+        [DataRow(@"HKU\.DEFAULT\Environment", "HKEY_USERS")]
+        public void GetRegistryBaseKeyTestOnlyOneBaseKey(string query, string expectedBaseKey)
+        {
+            var (baseKeyList, _) = RegistryHelper.GetRegistryBaseKey(query);
+            Assert.IsTrue(baseKeyList.Count() == 1);
+            Assert.AreEqual(expectedBaseKey, baseKeyList.FirstOrDefault().Name);
+        }
+
+        [TestMethod]
+        public void GetRegistryBaseKeyTestMoreThanOneBaseKey()
+        {
+            var (baseKeyList, _) = RegistryHelper.GetRegistryBaseKey("HKC\\Control Panel\\Accessibility");
```

Comment:
Same sort of thing wrt `HKC` ... although, maybe just add a `/* #no-spell-check-line */` to the end of the line and then add:
```
^.*/\* #no-spell-check-line \*/.*$
```
to `patterns.txt`
--- I haven't really standardized on a marker for this purpose, but this is the example pattern in https://github.com/check-spelling/check-spelling/wiki/Configuration-Examples:-patterns

`expect.txt` is a compromise between things which are always correct (which technically can live in `allow.txt`) and things which are transient, but which hopefully aren't easy typos for more correct things. For things that are exceptionally acceptable in specific instances, it's best to use `patterns.txt`, otherwise, in other cases where they're typos, the presence in `expect.txt` will result in the typo forms sneaking in.